tools/ctl/ide/explorer/explorer.go

```diff
@@ -399,6 +401,58 @@ func (e *Explorer) makeVerdictGroupNode(vg *rundex.VerdictGroup, percent float32
 	node.SetSelectedFunc(func() {
 		children := node.GetChildren()
 		if len(children) == 0 {
+			node.AddChild(tview.NewTreeNode("Find pattern").SetColor(tcell.ColorDarkCyan).SetSelectedFunc(func() {
+				go func() {
+					pattern := tview.NewTextArea()
+					var exitFunc func()
+					pattern.SetBorder(true).SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+						if event.Key() == tcell.KeyEnter {
+							input := pattern.GetText()
+							log.Printf("Finding pattern \"%s\"", input)
+							exitFunc()
+							regex, err := regexp.Compile(input)
+							if err != nil {
+								log.Println(err.Error())
+								return nil
+							}
+							go func() {
+								var found int
+								for _, example := range vg.Examples {
+									logs, err := e.butler.Fetch(context.Background(), example.RunID, example.WasSmoketest(), rebuild.DebugLogsAsset.For(example.Target()))
+									if err != nil {
+										log.Println(errors.Wrap(err, "downloading logs"))
+										continue
+									}
+									file, err := os.Open(logs)
+									if err != nil {
+										log.Println(errors.Wrap(err, "opening logs"))
+										continue
+									}
+									defer file.Close()
+									scanner := bufio.NewScanner(file)
+									for scanner.Scan() {
```

Comment:
i'd expect line-by-line to be slower but i'm willing to be shown otherwise.

one advantage of using a full-file buffer would be it would be easier to extract adjacent lines to provide context for each match (not sure how useful that would be but it might be something)